### ISSUE TITLE
Source generators

### ISSUE THREAD

**Issue by nicomoya123** [ekonbenefits/impromptu-interface#45]
I want to implement something similar to this package using the new source generators package,
Do you think it will be possible ?

**Comment by jbtule** [comment]
Yeah I think it's possible, for the explicit generic argument ActLike. 

As making the proxies is trivial source wise. 
just an object implementing an interface where each method is just forwarding the call to a dynamic typed _target.
Just have to identify each place you use it.

**Comment by goyzhang** [comment]
What if a method is not implemented in the target class. In that case, ActLike can compile and run as long as you don't call it. But the source generator won't compile and you need to have an exact match?
